### PR TITLE
GAWB-3668: Fix mc i/o autosuggest bug

### PR DESCRIPTION
**Pull request by ansingh7115**
This bug happened as result of trying to make the i/o Autosuggests cache - meaning that any addition of characters to those text boxes would trigger re-render. I reverted caching to false to fix this. In order to be able to assign values to the those text boxes (when populating via the "populate defaults" or via inputs json), I added refs to each Autosuggest and, mapping over each i/o that needs to be assigned, called `:set-value` on each Autosuggest. I had to add in `after-update` because when hitting either of the "populate" buttons, it will send you into edit mode (which renders the Autosuggests) and I need that render to happen _before_ I try to get the refs of the Autosuggests. 
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by ansingh7115** [comment]
jenkins retest

**Comment by ansingh7115** [comment]
jenkins retest

File: src/cljs/main/broadfcui/common/method/config_io.cljs [hunk 1]
```diff
@@ -51,20 +51,21 @@
...
-           {:keys [outputs editing?]} @state
+           {:keys [editing?]} @state
            new-outputs (->> (this :-get-defaultable-outputs)
                             (map (fn [{:keys [name]}] [(keyword name) (str "this." (-> name (string/split ".") last))]))
-                            (into {})
-                            (merge outputs))]
-       (when-not editing? (begin-editing))
-       (swap! state update :outputs merge new-outputs)))
+                            (into {}))
+           new-outputs-vector (into [] new-outputs)]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/method/config_io.cljs]: `(into [] ...)` can be `(vec ...)`, I think.

File: src/cljs/main/broadfcui/common/method/config_io.cljs [hunk 2]
```diff
@@ -51,20 +51,21 @@
...
+       (do (when-not editing? (begin-editing))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/method/config_io.cljs]: You don't need this `do`, the `let` will run everything.

File: src/cljs/main/broadfcui/common/method/config_io.cljs [hunk 3]
```diff
@@ -51,20 +51,21 @@
...
+           (after-update #(doseq [output new-outputs-vector] (((key output) @locals) :set-value (val output))))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common/method/config_io.cljs]: You could destructure `output` in the binding:
```clojure
(doseq [[k v] new-outputs-vector]
   ((k @locals) :set-value v))
```
And you should be able to `doseq` across a map in the same way, so you don't need to convert to a vector first.

**Review by MattPutnam — APPROVED** [review]
👍 with some tweaks, same thing on the input side of course.

File: src/cljs/main/broadfcui/common/method/config_io.cljs (hunk 3 above)
Comment (ansingh7115) [reply src/cljs/main/broadfcui/common/method/config_io.cljs]: that looks so much better! :D

**Review by ahaessly — APPROVED** [review]
👍

**Pull request merged**